performs multi word search. only displays common urls sorted by page …

PushkarBettadpur opened the pull request:
…rank
@harsh376 combined urls only

PushkarBettadpur committed:
performs multi word search. only displays common urls sorted by page rank

PushkarBettadpur commented:
Related to #12

harsh376 commented on run.py:
```diff
@@ -139,11 +144,21 @@ def home():
-
-        # Fetch relevant URLs ordered by page rank
+        words = get_all_words(search_string)
+        words = [i for i in words if i.lower() not in ignored_words]
```
replace `i.lower()` with `i`.

`parse_string` already converts to lower case

harsh376 commented on run.py:
```diff
@@ -139,11 +144,21 @@ def home():
+        for word in words:
+            wlist = search_db(db_conn=db_conn, word=word)
+            if orderedURLS is not None: 
+                orderedURLS = list(set(orderedURLS).intersection(wlist))
```
what happens if there are no common urls?

PushkarBettadpur replied:
Ok so what I'll do is, if there are no common urls I'll get all the links or the Union (rather than intersection) and sort it by rank.
We can reason our implementation in the report.

harsh376 replied:
yep. i think the order should be:
common urls   --------->  union of urls (excluding the ones that were common)

PushkarBettadpur committed:
Modifies common word search

If there are no common links (intersection is NULL), it finds the union of all links of all words
and also displays the words with no links in the results page

PushkarBettadpur commented:
@harsh376 check it out now. added a new commit

PushkarBettadpur committed:
rebased with master. changed num entries to 5 per page

PushkarBettadpur committed:
So now it's common links followed by highest ranked invidivual links. Also fixed a typo
